### PR TITLE
Token cli commands

### PR DESCRIPTION
**Pull request by Timo614**
## Description
Adds token CLI commands based on our discussion re Demo.

Was hoping to get the instruction commands working by having them send a request to the API. This way we'd use the same template runner / runtime for the processing there instead of having things run via the CLI. Having some trouble getting it to work though so confirming if I have the right format / blocked a bit on that front so left it somewhat empty in the PR.

We can either merge this as-is and iterate or keep it open and work in progress. 

## Motivation and Context
We're trying to add some additional CLI command support to spruce up the demo capabilities.

## How Has This Been Tested?
I've added a unit test for the new model method and have been trying out the CLI to test the new commands there.

List tokens for a given asset ID
```
cargo run -- token list 0000000100000000000000000000000.00000000000000000000000000000C73
Token -> List { asset_id: AssetID { template_id: TemplateID { template_type: 1, template_version: 0, tail: 0 }, features: 0, raid_id: RaidID("000000000000000"), hash: "00000000000000000000000000000C73" } }


Tokens of asset ID 0000000100000000000000000000000.00000000000000000000000000000C73
Id                                                                                               IssueNumber          Status

0000000100000000000000000000000.00000000000000000000000000000C736F5D7C2CA46D11EA8048000102030405 1                    Available
```

View token
```
cargo run -- token view 0000000100000000000000000000000.00000000000000000000000000000C736F5D7C2CA46D11EA8048000102030405
Token -> View { token_id: TokenID { asset_id: AssetID { template_id: TemplateID { template_type: 1, template_version: 0, tail: 0 }, features: 0, raid_id: RaidID("000000000000000"), hash: "00000000000000000000000000000C73" }, uid: 6f5d7c2c-a46d-11ea-8048-000102030405 } }


Token details:
Field                     Value

additional_data_json      {}
created_at                "2020-06-02T01:11:50.256195Z"
initial_data_json         {}
issue_number              1
status                    "Available"
token_id                  "0000000100000000000000000000000.0000000000000000000000
updated_at                "2020-06-02T01:11:50.256195Z"
```

Run instruction (WIP, having some issues locally hitting the actix endpoint / probably some issue with how I am approaching it but confirming with Maxim)
```
cargo run -- instruction asset 0000000100000000000000000000000.00000000000000000000000000000C73 issue_tokens '{"number": 6}'
Instruction -> Asset { asset_id: AssetID { template_id: TemplateID { template_type: 1, template_version: 0, tail: 0 }, features: 0, raid_id: RaidID("000000000000000"), hash: "00000000000000000000000000000C73" }, contract_name: "issue_tokens", data: Object({"number": Number(6)}) }
```

## Types of changes

* [ ] Bug fix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [x] New Tests
* [x] Documentation

## Checklist:

* [x] I'm merging against the `development` branch.
* [x] I ran `cargo-fmt --all` before pushing.
* [x] My change requires a change to the documentation.
* [x] I have updated the documentation accordingly.
* [x] I have added tests to cover my changes.

### CODE REVIEW

File: node/src/api/server.rs [hunk 1]
```diff
@@ -57,7 +57,8 @@ pub async fn actix_main(config: NodeConfig) -> anyhow::Result<()> {
                     .finish()
             })
             .wrap(Logger::new(LOGGER_FORMAT).exclude("/status"))
-            .wrap(Authentication::new())
+            // TODO: Should we not be using a JWT but rather something more custom?
+            //.wrap(Authentication::new())
```
Comment (Timo614) [inline node/src/api/server.rs:61]: I've disabled the authentication for now as we likely will want to change how we handle this. I imagine we want to use something more custom than the jwt algorithms for the signed token.

File: node/src/api/controllers/status.rs [hunk 2]
```diff
@@ -22,7 +23,7 @@ impl From<DeadpoolStatus> for Status {
     }
 }
 
-pub async fn check(db: Data<Pool>) -> Result<HttpResponse, ApiError> {
+pub async fn check(db: Data<Arc<Pool>>) -> Result<HttpResponse, ApiError> {
```
Comment (dunnock) [inline node/src/api/controllers/status.rs:26]: Ah I see, missed that endpoint when was converting pool to Arc

File: node/src/api/server.rs (hunk 1 above)
Comment (dunnock) [reply node/src/api/server.rs:61]: Yeah, makes sense

**Review by dunnock — APPROVED** [review]
it's all cool, going to merge that in myself now. Will look into bad request straight after

**Pull request merged**